pkg/controller/machinepool/nutanixactuator.go

```diff
@@ -0,0 +1,118 @@
+	scheme *runtime.Scheme
+	//nutanixClient *nutanixclient.Client
+}
+
+var _ Actuator = &NutanixActuator{}
+
+// NewNutanixActuator is the constructor for building a NutanixActuator
+func NewNutanixActuator(client client.Client, scheme *runtime.Scheme, logger log.FieldLogger) (*NutanixActuator, error) {
+	//nutanixClient, err := nutanixclient.NewClient()
+	//if err != nil {
+	//	return nil, errors.Wrap(err, "failed to create Nutanix session")
+	//}
+
+	actuator := &NutanixActuator{
+		client: client,
+		scheme: scheme,
+		//nutanixClient: nutanixClient,
+		logger: logger,
+	}
+
+	return actuator, nil
+}
+
+// GenerateMachineSets satisfies the Actuator interface and will take a clusterDeployment and return a list of MachineSets
+// to sync to the remote cluster.
+func (a *NutanixActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *hivev1.MachinePool, logger log.FieldLogger) ([]*machineapi.MachineSet, bool, error) {
+	if cd.Spec.ClusterMetadata == nil {
+		return nil, false, errors.New("ClusterDeployment does not have cluster metadata")
+	}
+
+	if cd.Spec.Platform.Nutanix == nil {
+		return nil, false, errors.New("ClusterDeployment is not for Nutanix")
+	}
+	if pool.Spec.Platform.Nutanix == nil {
+		return nil, false, errors.New("MachinePool is not for Nutanix")
+	}
+
+	computePool := baseMachinePool(pool)
+
+	var disks []installertypesnutanix.DataDisk
+	for deviceIndex, disk := range pool.Spec.Platform.Nutanix.Disks {
+		disks = append(disks, installertypesnutanix.DataDisk{
+			DiskSize: resource.MustParse(fmt.Sprintf("%db", disk.DiskSizeBytes)),
+			DeviceProperties: &machinev1.NutanixVMDiskDeviceProperties{
+				DeviceType:  disk.DeviceType,
+				AdapterType: disk.AdapterType,
+				DeviceIndex: int32(deviceIndex),
+			},
+			StorageConfig:   nil,
+			DataSourceImage: nil,
+		})
+	}
+
+	computePool.Platform.Nutanix = &installertypesnutanix.MachinePool{
+		NumCPUs:           pool.Spec.Platform.Nutanix.NumSockets,
+		NumCoresPerSocket: pool.Spec.Platform.Nutanix.NumVcpusPerSocket,
+		MemoryMiB:         pool.Spec.Platform.Nutanix.MemorySizeMiB,
+		BootType:          pool.Spec.Platform.Nutanix.BootType,
+		DataDisks:         disks,
+	}
+
+	ic := &installertypes.InstallConfig{
+		Platform: installertypes.Platform{
+			Nutanix: &installertypesnutanix.Platform{
+				PrismCentral: installertypesnutanix.PrismCentral{
+					Endpoint: installertypesnutanix.PrismEndpoint{
+						Address: cd.Spec.Platform.Nutanix.Endpoint,
+						Port:    cd.Spec.Platform.Nutanix.Port,
+					},
+					Username: "",
+					Password: "",
+				},
+				PrismElements:  nil,
+				IngressVIPs:    nil,
+				SubnetUUIDs:    []string{cd.Spec.Platform.Nutanix.Subnet},
+				FailureDomains: nil,
+			},
+		},
+	}
+
+	installerMachineSets, err := installvsphere.MachineSets(
+		cd.Spec.ClusterMetadata.InfraID,
+		ic,
+		computePool,
+		// With FailureDomains[0].Topology.Template set, this param is ignored.
+		"HIVE_BUG!",
+		workerRole,
+		workerUserDataName,
+	)
```

Comment:
copypasta, looks like this still needs some work.

When we get down to it though, I think we should take the nutanix installer team up on their offer to help out by reviewing this func.

The general principle is that our code here needs to dummy up the inputs to the platform-specific MachineSets generator. We can not rely on having access to the original install-config.yaml for the cluster, which is sometimes annoying and awkward, forcing us to either assume certain values or duplicate them in the (hive) MachinePool CRD. Since we're making this actuator from scratch, we should take the opportunity to evaluate those things carefully; else we can end up painting ourselves into a corner where e.g. we have to support bogus/deprecated/obsolete fields "forever".